src/sentry/static/sentry/app/stores/modalStore.tsx

```diff
@@ -25,6 +25,13 @@ const ModalStore = Reflux.createStore({
   },
 
   onCloseModal() {
+    const onClose = this.state?.options?.onClose;
+
+    // Trigger the options.onClose callback
+    if (typeof onClose === 'function') {
+      onClose();
+    }
```

Comment:
This doesn't feel great, but neither does using bootstrap modals.

Reply:
The issue is we're inconsistent right now, it's called when you click outside of the modal, but now when you trigger the close action of the modal.

Reply:
Seems like good info to have in the PR body.